armi/operators/settingsValidation.py

```diff
@@ -726,3 +727,45 @@ def createQueryRevertBadPathToDefault(inspector, settingName, initialLambda=None
         inspector.cs.getSetting(settingName).revertToDefault,
     )
     return query
+
+
+def validateVersion(versionThis: str, versionRequired: str) -> bool:
+    """Helper function to allow users to verify that their version matches the settings file.
+
+    Parameters
+    ----------
+    versionThis: str
+        The version of this ARMI, App, or Plugin.
+        This MUST be in the form: 1.2.3
+    versionRequired: str
+        The version to compare against, say in a Settings file.
+        This must be in one of the forms: 1.2.3, 1.2, or 1
+
+    Returns
+    -------
+    bool
+        Does this version match the version in the Settings file/object?
+    """
+    fullV = "[0-9]\.[0-9]\.[0-9]"
+    medV = "[0-9]\.[0-9]"
+    minV = "[0-9]"
```

Comment:
I think these patterns need to be extended to handle numbers with multiple digits?

Reply:
AH crap.

Yes.  You're right. Dammit. Thanks!

Reply:
Fixed!